File: pset6/caesar/caesar.py
from cs50 import get_string
import sys

# TO ENSURE CORRECT USAGE
if(len(sys.argv) != 2):
    print('Usage: python caesar.py key')
    exit(1)

print("plaintext: ", end="")
plaintext = get_string()

# TO CONVERT THE USER INPUTED KEY INTO INT
key = int(sys.argv[1])

ciphertext = ""
ckey = 0

# TO ITERATE OVER EACH CHARACTER IN plaintext
for c in (plaintext):
    ckey = ord(c) + (key)
    # IN THE CASE OF LOWER CASE LETTERS
    if c.islower():
        if ckey > 122:
            ciphertext += chr(97 + ((ckey - 97) % 26))
        else:
            ciphertext += chr(ckey)
    # IN THE CASE OF UPPER CASE LETTERS
    elif c.isupper():
        if ckey > 90:
            ciphertext += chr(65 + ((ckey - 65) % 26))
        else:
            ciphertext += chr(ckey)
    # IN THE CASE OF NEITHER LOWER CASE NOR UPPER CASE LETTERS
    else:
        ciphertext += c

print("ciphertext: ", ciphertext)

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>


int main(int argc,string argv[])
{
    if(argc==2)
    {
    int key=atoi(argv[1]);
    printf("plaintext: ");
    string plain_text=get_string();
    int len=strlen(plain_text);
    printf("ciphertext: ");
    int c;
    for(int i=0;i<len;i++)
    {
        c=key+plain_text[i];
        if(plain_text[i]>=65 && plain_text[i]<=90)
        {
            if(c>90)
            {
                printf("%c",65+((c-65)%26));
            }
            else
            {
                printf("%c",c);
            }
        }
        else if(plain_text[i]>=97 && plain_text[i]<=122)
        {
            if(c>122)
            {
            printf("%c",97+((c-97)%26));
            }
            else
            {
                printf("%c",c);
            }
        }
        else
        {
            printf("%c",plain_text[i]);
        }
    }
    printf("\n");
    }
    else
    {
        printf("retry");
        return 1;
    }
}